### PR TITLE
140 - Rider campaign signup

### LINKED ISSUE #140: Riders sign up for a specific campaign via app
Mocks for this https://xd.adobe.com/view/e64dceb0-2df2-400f-8971-405ce662631f-b31e/screen/ef229902-9870-4af9-8024-77d66ec81fd7/

Dispatcher side at #139

### PR DESCRIPTION
**Pull request by teesloane**
Closes #140. Commits are atomic and go into more detail. 

https://github.com/bikebrigade/dispatch/assets/12987958/d39ed6b2-fb32-4292-8d95-42ae93985cd9

### CODE REVIEW

File: lib/bike_brigade_web/live/campaign_signup_live/show.html.heex [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+<section>
+  <.table id="tasks" rows={@tasks}>
+    <:col :let={_task} label="Pickup time">
+      <%= pickup_window(@campaign) %>
+    </:col>
+    <:col :let={task} label="Delivery Size">
+      <.get_delivery_size task={task}/>
+    </:col>
+    <:col :let={task} label="Recipient"><%= task.dropoff_name %></:col>
+    <:col :let={task} label="Dropoff Location"><%= task.dropoff_location.address %></:col>
```
Comment (teesloane) [inline lib/bike_brigade_web/live/campaign_signup_live/show.html.heex:20]: Right now this shows the exact address of the Task. Is this what we want?

File: lib/bike_brigade_web/live/campaign_signup_live/show.ex [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    :task_deleted,
+    :task_updated,
+    :campaign_rider_created,
+    :campaign_rider_deleted
+  ]
+
+  @impl Phoenix.LiveView
+  def handle_info({event, entity}, socket) when event in @broadcasted_infos do
+    campaign = socket.assigns.campaign
+    # if a task or a campaign rider changes (ie, if any of hte broadcasted_infos)
```
Comment (mveytsman) [inline lib/bike_brigade_web/live/campaign_signup_live/show.ex:125]: typo

**Review by mveytsman — CHANGES_REQUESTED** [review]
* When singing up, riders shouldn't see exact location of the task
	- Switch to showing neighbourhood or abstracted location 
* bug about navigating via "back button" when rider signs up for themselves
* leave a note on `campaign_signup_live/index.ex` about the database fetching
* "Campaign Filled" should be no colour?
* Remove rider signup form - so just click and it does it.
* Remove "capacity" and "pickup window" from rider signup form
	* Maybe remove enter buildings
* consider adding a rider setting that enables people to choose how much of their PI is shown when singing up
	- nick name, full name, etc

**Review by mveytsman — APPROVED** [review]
🎉

**Pull request merged**